### PR TITLE
feat: Add ODBC Databricks Benches

### PR DESCRIPTION
**Pull request by peasee**
## 🗣 Description

* Adds benchmark tests for ODBC Databricks, configured for the AWS Databricks instance from repo secrets

Benches pass locally:

```
+--------------------------------------+---------------+---------------+----------------+--------+-----------------+-----------------+------------+----------------+-------------+----------------+
| run_id                               | started_at    | finished_at   | query_name     | status | min_duration_ms | max_duration_ms | iterations | commit_sha     | branch_name | connector_name |
+--------------------------------------+---------------+---------------+----------------+--------+-----------------+-----------------+------------+----------------+-------------+----------------+
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951360291 | 1721951390757 | tpch_q1        | passed | 5711            | 6387            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951390757 | 1721951429702 | tpch_q2        | passed | 6128            | 12162           | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951429702 | 1721951482073 | tpch_q3        | passed | 9305            | 12650           | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951482074 | 1721951506545 | tpch_q4        | passed | 3984            | 6451            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951506546 | 1721951565177 | tpch_q5        | passed | 6609            | 14171           | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951565177 | 1721951581634 | tpch_q6        | passed | 3238            | 3464            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951581634 | 1721951630554 | tpch_q7        | passed | 9510            | 9974            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951630554 | 1721951665357 | tpch_q8        | passed | 6171            | 9206            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951665357 | 1721951715388 | tpch_q9        | passed | 9761            | 10381           | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951715388 | 1721951822565 | tpch_q10       | passed | 17198           | 25996           | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951822565 | 1721951839624 | tpch_q11       | passed | 3151            | 3833            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951839624 | 1721951858298 | tpch_q12       | passed | 3520            | 4259            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951858298 | 1721951890930 | tpch_q13       | passed | 5811            | 8523            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951890930 | 1721951909609 | tpch_q14       | passed | 3589            | 4027            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951909609 | 1721951937504 | tpch_q16       | passed | 5346            | 5985            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951937504 | 1721951963922 | tpch_q17       | passed | 4396            | 8399            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721951963922 | 1721952038008 | tpch_q18       | passed | 14267           | 15401           | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721952038008 | 1721952058344 | tpch_q19       | passed | 3760            | 4788            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721952058344 | 1721952089691 | tpch_q20       | passed | 6078            | 6592            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721952089692 | 1721952169959 | tpch_q21       | passed | 15472           | 17745           | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721952169959 | 1721952189303 | tpch_q22       | passed | 3718            | 4216            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721952189303 | 1721952202396 | tpch_simple_q1 | passed | 2481            | 2999            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721952202396 | 1721952330836 | tpch_simple_q2 | passed | 25220           | 26207           | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721952330836 | 1721952356968 | tpch_simple_q3 | passed | 4097            | 8457            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721952356968 | 1721952373557 | tpch_simple_q4 | passed | 2786            | 5047            | 5          | 32b8b778-dirty | trunk       | odbc           |
| 254483d8-3ee1-4f6a-aee1-dc7ab11b4b50 | 1721952373557 | 1721952420622 | tpch_simple_q5 | passed | 9156            | 10101           | 5          | 32b8b778-dirty | trunk       | odbc           |
+--------------------------------------+---------------+---------------+----------------+--------+-----------------+-----------------+------------+----------------+-------------+----------------+
```

### CODE REVIEW

**Comment by peasee** [comment]
actions test run in progress: https://github.com/spiceai/spiceai/actions/runs/10103382646

**Comment by peasee** [comment]
need to get the databricks odbc driver installed in actions

**Comment by peasee** [comment]
passing actions benchmark run: https://github.com/spiceai/spiceai/actions/runs/10103534053

**Review by sgrebnov — DISMISSED** [review]

File: crates/runtime/benches/bench_odbc_databricks/mod.rs [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+use spicepod::component::{dataset::Dataset, params::Params};
+
+pub(crate) async fn run(
+    rt: &mut Runtime,
+    benchmark_results: &mut BenchmarkResultsBuilder,
+) -> Result<(), String> {
+    let test_queries = get_test_queries();
+
+    for (query_name, query) in test_queries {
+        super::run_query_and_record_result(rt, benchmark_results, "odbc", query_name, query)
```
Comment (sgrebnov) [inline crates/runtime/benches/bench_odbc_databricks/mod.rs]:
```suggestion
        super::run_query_and_record_result(rt, benchmark_results, "odbc-databricks", query_name, query)
```

**Review by phillipleblanc — APPROVED** [review]

**Pull request merged**